### PR TITLE
Google Optimize BugFix : Scriptloader to add script inside head

### PR DESCRIPTION
**Pull request by abandonned**
## Description of the change
Updated the default scriptloader to add the script inside head.

https://www.notion.so/Google-Optimize-Web-Device-Mode-Always-add-GO-script-to-the-head-348f524b540749178b7ca4a66f6f8e30

## Type of change
- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Related issues

> Fix [#1]() 

### Development

- [ ] Lint rules pass locally
- [ ] The code changed/added as part of this pull request has been covered with tests
- [ ] All tests related to the changed code pass in development

### Code review 

- [ ]  This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached
- [ ] "Ready for review" label attached to the PR and reviewers mentioned in a comment
- [ ] Changes have been reviewed by at least one other engineer
- [ ] Issue from task tracker has a link to this pull request

---
This change is [Reviewable](https://reviewable.io/reviews/rudderlabs/rudder-sdk-js/424)

### CODE REVIEW

File: dist/rudder-sdk-js/package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "rudder-sdk-js",
-  "version": "1.2.15",
+  "version": "1.2.19",
```
Comment (saikumarrs) [inline dist/rudder-sdk-js/package.json:3]: Why is this change here?

File: integrations/GoogleOptimize/util.js [hunk 2]
```diff
@@ -0,0 +1,13 @@
+const ScriptLoaderHead = (id, src, async = defaultAsyncState) => {
```
Comment (saikumarrs) [inline integrations/GoogleOptimize/util.js:1]: Let's update the common script loader utility

File: dist/rudder-sdk-js/package.json (hunk 1 above)
Comment (abandonned) [reply dist/rudder-sdk-js/package.json:3]: I created branch from production.

File: integrations/GoogleOptimize/util.js [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+    logger.debug(`in script loader=== ${id}`);
+    const js = document.createElement("script");
+    js.src = src;
+    js.async = async === undefined ? defaultAsyncState : async;
+    js.type = "text/javascript";
+    js.id = id;
+    const e = document.getElementsByTagName("head")[0];
+    logger.debug("==adding script==", js);
+    e.appendChild(js);
```
Comment (saikumarrs) [inline integrations/GoogleOptimize/util.js:10]: It is always recommended to add the new script to the beginning of the existing ones.

Comment (abandonned) [reply integrations/GoogleOptimize/util.js:10]: `const h = document.getElementsByTagName("head")[0];
  const e = h.getElementsByTagName("script")[0];
  e.parentNode.insertBefore(js, e);`
like this?

Comment (saikumarrs) [reply integrations/GoogleOptimize/util.js:10]: Yeah, similar to what we do in the common script loader utility

Comment (saikumarrs) [reply integrations/GoogleOptimize/util.js:10]: Although, you wouldn't need `e.parentNode` rather it should simply be `h.insertBefore(js, e)`

Comment (abandonned) [reply integrations/GoogleOptimize/util.js]: updated

File: integrations/GoogleOptimize/util.js (hunk 2 above)
Comment (abandonned) [reply integrations/GoogleOptimize/util.js]: updated

File: integrations/ScriptLoader.js [hunk 4]
```diff
@@ -9,10 +10,11 @@ const ScriptLoader = (id, src, async = defaultAsyncState) => {
   js.async = async === undefined ? defaultAsyncState : async;
   js.type = "text/javascript";
   js.id = id;
-  const e = document.getElementsByTagName("script")[0];
+  const h = document.getElementsByTagName("head")[0];
+  const e = h.getElementsByTagName("script")[0];
   logger.debug("==parent script==", e);
   logger.debug("==adding script==", js);
-  e.parentNode.insertBefore(js, e);
+  h.insertBefore(js, e);
```
Comment (saikumarrs) [inline integrations/ScriptLoader.js:17]: Let's take this approach,
```if <head> tag is available:
  // use it to add the new script
else
  // fallback to the original implementation```
